### PR TITLE
Macro infinite recursion

### PR DESCRIPTION
**Pull request by bruntib**

### CODE REVIEW

File: test/Analysis/Inputs/expected-plists/plist-macros-with-expansion.cpp.plist [hunk 1]
```diff
@@ -5443,6 +5443,140 @@
...
+   <array>
+    <dict>
+     <key>location</key>
+     <dict>
+      <key>line</key><integer>450</integer>
+      <key>col</key><integer>7</integer>
+      <key>file</key><integer>0</integer>
+     </dict>
+     <key>name</key><string>value</string>
+     <key>expansion</key><string>garbage_</string>
```
Comment (Szelethus) [inline test/Analysis/Inputs/expected-plists/plist-macros-with-expansion.cpp.plist]: I mean, yea. _Proceeds to drop an F-bomb_

Here's the deal with edge cases: Sure, they are incredibly rare, but most of the time, incredibly rare cases like this tend to hide the nastiest bugs, because, well, they are extremely rare and hard to follow what will actually happen. If you could, please, try to see whether it's feasible to provide a proper expansion to this. I am open for negotiations on just letting this go, but would prefer not to.
